pysmt/smtlib/script.py

```diff
@@ -77,16 +77,53 @@ def serialize(self, outstream=None, printer=None, daggify=True):
             printer.printer(self.args[0])
             outstream.write(")")
 
+        elif self.name == smtcmd.ASSERT_SOFT:
+            outstream.write("(%s " % self.name)
+            printer.printer(self.args[0])
+            for arg in self.args[1]:
+                option_name, value = arg
+                if option_name == ":weight":
+                    outstream.write(" %s " % option_name)
+                    printer.printer(value)
+                else:
+                    outstream.write(" %s %s" % (option_name, value))
+            outstream.write(")")
+
+
         elif self.name == smtcmd.GET_VALUE:
             outstream.write("(%s (" % self.name)
             for a in self.args:
                 printer.printer(a)
                 outstream.write(" ")
             outstream.write("))")
 
+        elif self.name in [smtcmd.MAXIMIZE, smtcmd.MINIMIZE]:
+            outstream.write("(%s " % self.name)
+            printer.printer(self.args[0])
+            for arg in self.args[1]:
```

Comment:
Can we use a different variable name? I would expect arg to be a member of `self.args`, not of `self.args[i]`. I think above we used `a`. In this case this would also be consistent as we are talking about annotations.